File: src/js/tests/BlockquoteTests.js
var blockquoteTests = function() {
	var parser;
	
	return {
		setup: function() {
			parser = createParser();
		},
		
		tearDown: function() {
			parser = null;
		},
		
		test: function Blockquote_tests() {
			return [
				function Should_parse_blockquote_on_same_line() {
					Assert.that(trimLf(parser.parseAndReturn("<<text>>")), Is.equalTo("<blockquote><p>text</p></blockquote>"));	
				},
				
				function Should_parse_blockquote_on_different_line() {
					Assert.that(trimLf(parser.parseAndReturn("<<\ntext\n>>")), Is.equalTo("<blockquote><p>text</p></blockquote>"));	
				},
				
				function Should_allow_formatting_inside_blockquote() {
					Assert.that(trimLf(parser.parseAndReturn("<<oh __hai!__>>")), Is.equalTo("<blockquote><p>oh <strong>hai!</strong></p></blockquote>"));	
				},
				
				function Should_not_allow_blockquote_inside_inline_scope() {
					Assert.that(trimLf(parser.parseAndReturn("__foo\n<<not a blockquote\n__")), Is.equalTo("<p><strong>foo&lt;&lt;not a blockquote</strong></p>"));	
				}
				
			];
		}
	};
}();

Jarvis.run(blockquoteTests);

File: src/dotnet/Tests/ButterflyNet.Tests/BlockquoteTests.cs
using NUnit.Framework;

namespace ButterflyNet.Parser.Tests {
	[TestFixture]
	public class BlockquoteTests : WikiToHtmlTest {
		[Test]
		public void Should_parse_blockquote_on_same_line() {
			AssertWithNoRegardForLineBreaks(Parser.ParseAndReturn("<<text>>"), "<blockquote><p>text</p></blockquote>");
		}

		[Test]
		public void Should_parse_blockquote_on_different_line() {
			AssertWithNoRegardForLineBreaks(Parser.ParseAndReturn("<<\ntext\n>>"), "<blockquote><p>text</p></blockquote>");
		}

		[Test]
		public void Should_allow_formatting_inside_blockquote() {
			AssertWithNoRegardForLineBreaks(Parser.ParseAndReturn("<<oh __hai!__>>"), "<blockquote><p>oh <strong>hai!</strong></p></blockquote>");
		}

		[Test]
		public void Should_not_allow_blockquote_inside_inline_scope() {
			AssertWithNoRegardForLineBreaks(Parser.ParseAndReturn("__foo\n<<not a blockquote\n__"), "<p><strong>foo&lt;&lt;not a blockquote</strong></p>");
		}
	}
}